### PR TITLE
refactor: [M3-7509, M3-7518] TopMenu and TagsList v7 Stories

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
Small PR to get rid of more `.mdx` stories and replace them with a `.tsx` one when relevant.
In this PR we are refactoring TopMenu & Tags stories and getting rid of icons since they are not relevant anymore.

This does not touch any client facing code.

## Changes  🔄
- New v7 story for Tags List
- New v7 story for TopMenu
- Remove icons `.mdx` stories

## Preview 📷
Screenshot 2023-11-30 at 16 49 48

Screenshot 2023-11-30 at 16 50 00

### Verification steps 
pull code locally and run `yarn storybook`
- verify http://localhost:6006/?path=/docs/components-tags-tags-list--documentation
- verify http://localhost:6006/?path=/docs/features-navigation-top-menu--documentation

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/ColorPalette/ColorPalette.stories.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
 import { ColorPalette } from './ColorPalette';
 
-<Meta title="Core Styles/Color Palette" component={ColorPalette} />
+<Meta title="Design System/Color Palette" component={ColorPalette} />
```
Comment (abailly-akamai) [inline packages/manager/src/components/ColorPalette/ColorPalette.stories.mdx:4]: This is a bit more relevant a title, and will also push this category down. Eventually the design system will have its own storybook and it won't belong here, but for now we'll keep this page alive.

File: packages/manager/src/features/TopMenu/TopMenu.tsx [hunk 2]
```diff
@@ -17,13 +17,17 @@ import SearchBar from './SearchBar/SearchBar';
...
+export interface TopMenuProps {
   desktopMenuToggle: () => void;
   isSideMenuOpen: boolean;
   openSideMenu: () => void;
   username: string;
 }
 
+/**
+ * - Items presented in the top navigation are considered universally important and should be available regardless of any particular task.
+ * - The number of items should be limited. In the future, **Help & Support** could become a drop down with links to **Community**, **Guides**, and etc.
```
Comment (mjac0bs) [inline packages/manager/src/features/TopMenu/TopMenu.tsx:29]: > In the future, **Help & Support** could become a drop down with links to **Community**, **Guides**, and etc.

I know this was existing copy, just ported over. I wonder if we have a ticket for this.

**Review by mjac0bs — APPROVED** [review]
Stories looked good.

I agree with your retitling of the Color Palette story and deletion of the outdated icons.

Thanks for removing mdx!

**Review by coliu-akamai — APPROVED** [review]
Thanks for the cleanup! 
✅ confirmed stories look good!

**Pull request merged**